test/common/index.js

```diff
@@ -802,3 +802,25 @@ exports.hijackStdout = hijackStdWritable.bind(null, 'stdout');
 exports.hijackStderr = hijackStdWritable.bind(null, 'stderr');
 exports.restoreStdout = restoreWritable.bind(null, 'stdout');
 exports.restoreStderr = restoreWritable.bind(null, 'stderr');
+
+// Check if the executable links to the shared lib node or not.
+exports.isSharedLibNode = process.config.variables.node_shared === true;
+
+// If node executable is linked to shared lib, need to take care about the
+// shared lib path.
+exports.addLibraryPath = function() {
+  if (!process.config.variables.node_shared) {
+    return;
+  }
+  process.env.LD_LIBRARY_PATH =
+    path.join(path.dirname(process.execPath), 'lib.target');
```

Comment:
Should this extend `LD_LIBRARY_PATH` instead of replace it? (Same for `LIBPATH` and `DYLD_LIBRARY_PATH` below).

Reply:
Sure. Let me change to extend these variables.